File: Repository/Models/CurriculumVitae.cs
using Repository.Interfaces;

namespace Repository.Models;

public class CurriculumVitae : IIdentificator
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    // public DateTime CreationDate { get; set; }
    public DateTime ModificationDate { get; set; }

    public ApplicationUser ApplicationUser { get; set; }
    public string ApplicationUserId { get; set; }

    public PersonalData PersonalData { get; set; }
    public CvAddress CvAddress { get; set; }
    public ContactData ContactData { get; set; }

    public IEnumerable<CvWorkExperience> WorkExperience { get; set; }
    public IEnumerable<CvEducation> Education { get; set; }

    // public Guid PersonalDataId { get; set; }
}

File: CvCreator/src/app/Models/CurriculumVitae.ts
import { ContactData } from './ContactData';
import { Address } from './Address';
import { PersonalData } from './PersonalData';
import { WorkExperience } from './WorkExperience';
import { Education } from './Education';
import { Language } from './Language';
export class CurriculumVitae {
  id: string;
  name: string;
  creationDate: string;
  creationTime: string;

  personalData?: PersonalData;
  cvAddress?: Address;
  contactData?: ContactData;
  workExperience?: WorkExperience[]
  education?: Education[]
  languages?: Language[]
}
